### ISSUE TITLE
Cluster count is varying while shuffling the data set

### ISSUE THREAD

**Issue by jerilkuriakose** [scikit-learn-contrib/hdbscan#259]
The number of unique clusters are varying while shuffling the dataset. The following is my code:

    import hdbscan
    import numpy as np
    import sklearn.datasets as data
    def get_data():
        moons, _ = data.make_moons(n_samples=30000, noise=0.05)
        blobs, _ = data.make_blobs(n_samples=30000, centers=[(-0.75,2.25), (1.0, 2.0)], cluster_std=0.25)
        return moons, blobs
    moons, blobs = get_data()

The above code get the data, now I am trying to stack the data and shuffle the data

    test_data = np.vstack([moons, blobs])
    np.random.shuffle(test_data) # shuffling the dataset
    clusterer = hdbscan.HDBSCAN()
    clusterer.fit(test_data)
    print('Cluster count:', np.unique(clusterer.labels_).size)

While shuffling the dataset I am getting the number of unique clusters as `1169`, but when I am not shuffling the dataset I am getting the number of unique clusters as `596` . What might be the reason behind this? Kindly suggest.

**Comment by shex1627** [comment]
I think it has to do with the randomness of the algorithm instead of the shuffling. Whether you shuffle the data or not, if you run the code a few times, you may see cluster of 4-5, 500-600, ~1000.
